.eslintrc.cjs

```diff
@@ -20,6 +20,17 @@ module.exports = {
         'sourceType': 'script',
       },
     },
+    {
+      'files': ['**/*.ts'],
+      'parser': '@typescript-eslint/parser',
+      'extends': [
+        'plugin:@typescript-eslint/recommended',
+      ],
+      'rules': {
+        '@typescript-eslint/no-non-null-assertion': 2,
```

Comment:
How does eslint with typescript work here? Will the eslint rules also apply to the .ts files or the converted ones? Or do all the equivalent rules turned on for eslint need to be turned on for typescript as well?

Reply:
> How does eslint with typescript work here? Will the eslint rules also apply to the .ts files or the converted ones? Or do all the equivalent rules turned on for eslint need to be turned on for typescript as well?

It will merge rules from `@typescript-eslint/recommended` in to old js rules. We don't need to do it twice

Reply:
ESLint doesn't know JavaScript or TypeScript, it only knows AST(Abstract Syntax Tree). The `@typescript-eslint/parser` would parse `.ts` file to AST that compatiable with ESLint's. So ESLint can capture common syntax error (TypeScript claims it is a superset of JavaScript), but if that is a TypeScript-specific syntax, only `@typescript-eslint` could do the such thing.

Reply:
Thanks for the explanation.